libmscore/fret.cpp

```diff
@@ -539,10 +566,13 @@ void FretDiagram::writeOld(XmlWriter& xml) const
 
             for (auto const& d : allDots) {
                   if (d.exists()) {
-                        if (d.fret < lowestDotFret || lowestDotFret == -1)
+                        if (d.fret < lowestDotFret || lowestDotFret == -1) {
                               lowestDotFret = d.fret;
-                        else if (d.fret == lowestDotFret)
-                              multipleDotsOnFret = true; 
+                              furthestLeftLowestDot = i;
```

Comment:
I guess this is the code to fix the issue you mentioned writing barres, but I'd like to know what the issue was to help me test.

Reply:
Compatability for versions < 3.1. Barres can only be written in a handful of cases due to the terrible fret diagram writing format we had back in those versions. You could test it opening a score with barres from 3.1 in 3.0.5.